builder - evm template - generalization (#198)
moondance-labs/tanssi-docs: builders/build/templates/evm.md

## Introduction {: #introduction }

> Tanssi's EVM (Ethereum Virtual Machine) appchain template is specifically designed for teams developing their applications on top of an EVM (Ethereum Virtual Machine). It includes all the essential components needed to add an additional layer for full Ethereum compatibility:

Reviewer: ```suggestion
Tanssi's EVM (Ethereum Virtual Machine) appchain template is designed for teams developing their applications on top of EVM smart contracts. It includes all the essential components needed for a full Ethereum-compatible appchain:
```

- **EVM** - adds an Ethereum Virtual Machine execution layer for EVM-based smart contract applications
- **Etherum JSON RPC Support** - Tanssi EVM appchains are fully [Ethereum JSON RPC](https://ethereum.org/en/developers/docs/apis/json-rpc){target=\_blank} compliant. Consequently, all Ethereum-based tools like [MetaMask](https://metamask.io){target=\_blank}, [Ethers.js](https://docs.ethers.org){target=\_blank}, [Viem](https://viem.sh){target=\_blank}, [Hardhat](https://hardhat.org){target=\_blank}, [Foundry](https://book.getfoundry.sh){target=\_blank} and more, work seamlessly out of the box
- **Unified Accounts** - allows Tanssi EVM appchains to feature Ethereum-styled ECDSA accounts

## EVM Appchain Template {: #evm-appchain-template }

> Since the template already includes the necessary configuration for seamless integration with the Tanssi protocol and the security provider, if the use case is entirely developed on top of the EVM, then this template requires no additional changes in the runtime.

Reviewer: ```suggestion
The template already includes the necessary configuration for seamless integration with the Tanssi protocol and the security provider of choice, for example, [Symbiotic](https://symbiotic.fi/){target=\_blank} on Ethereum. Therefore, this template requires no additional changes in the runtime if the application is built on top of the EVM.

This means that this template is ready to be deployed as-is through Tanssi, unlocking many features, such as:

- Utilize Tanssi's [block production as a service](/learn/tanssi/technical-features/#block-production-as-a-service){target=\_blank}
- Get deterministic transaction finality in seconds

> - Leverage Ethereum-grade security from [Symbiotic](https://symbiotic.fi/){target=\_blank}

Reviewer: Same comment as before?

To reach full Ethereum compatibility, these specific modules are also included:

> - **[EVM module](https://docs.rs/pallet-evm/latest/pallet_evm){target=\_blank}** - it adds support for unmodified EVM bytecode execution on a Tanssi appchain. It uses the Rust-based [SputnikVM](https://github.com/rust-ethereum/evm){target=\_blank} as the underlying EVM engine

Reviewer: Same comment as before, if we are using modules at the top we can remove them here?
